src/pages/gd/Swap/SwapConfirmModal/index.tsx

```diff
@@ -65,28 +65,67 @@ function SwapConfirmModal({
     const web3 = useWeb3()
     const [status, setStatus] = useState<'PREVIEW' | 'CONFIRM' | 'SENT'>('SENT')
     const [hash, setHash] = useState('')
+    const [sender, setSender] = useState('')
+
+    const senderRef = useRef(sender)
+
+    useEffect(() => {
+      senderRef.current = sender
+    }, [sender])
 
     const handleSwap = async () => {
         setStatus('CONFIRM')
+
+        const prepareTx = (from: string) => {
+          setSender(from)
+        }
+        
         const onSent = (hash: string) => {
             setStatus('SENT')
             setHash(hash)
-            if (onConfirm) onConfirm()
-        }
-        try {
-            let transactionDetails = buying ? await buy(web3!, meta!, onSent) : await sell(web3!, meta!, onSent)
+            const inputSig = meta?.inputAmount.toSignificant(5)
+            const minimumOutputSig = meta?.minimumOutputAmount.toSignificant(5)
+
+            const tradeInfo = {
+              input: {
+                decimals: meta?.inputAmount.currency.decimals,
+                symbol: meta?.inputAmount.currency.symbol
+              },
+              output: {
+                decimals: meta?.outputAmount.currency.decimals,
+                symbol: meta?.outputAmount.currency.symbol
+              }
+            }
+            const summary = 'Swapping ' + inputSig + ' ' 
+                            + meta?.inputAmount.currency.symbol + 
+                            ' to a minimum of ' + minimumOutputSig + ' ' +
+                            meta?.outputAmount.currency.symbol
             globalDispatch(
-                addTransaction({
-                    chainId: chainId!,
-                    hash: transactionDetails.transactionHash,
-                    from: transactionDetails.from
-                })
+              addTransaction({
+                chainId: chainId!,
+                hash: hash,
+                from: senderRef.current,
+                summary: summary,
+                tradeInfo: tradeInfo,
+              })
             )
+            if (onConfirm) onConfirm()
+        }
+        try { 
+            buying ? await buy(web3!, meta!, prepareTx, onSent) : await sell(web3!, meta!, prepareTx, onSent)
+            // let transactionDetails = buying ? await buy(web3!, meta!, prepareTx, onSent) : await sell(web3!, meta!, prepareTx, onSent)
+            // globalDispatch(
+            //     addTransaction({
+            //         chainId: chainId!,
+            //         hash: transactionDetails.transactionHash,
+            //         from: transactionDetails.from
+            //     })
+            // )
```

Comment:
why was this replaced? it looks more correct to me... to add the transaction after it is sent to the blockchain

Reply:
That is what is happening now right?
onSent is triggered when transaction hash is available
the return of buy is after the last event, so with the old flow it only added an already confirmed transaction
(so then the flow of transaction actions (add/checked/finalized) is not triggered properly as it never adds a pending transaction)

or should I consider removing the onSent logic I added, and just do everything in buy/sell?
So that the only thing onSent triggers is setStatus('SENT') (for the Transaction submitted dialog message)
though other comments considering, having it here makes it more DRY?

Reply:
no you are right, I didnt notice your changes are part of the onSent

Reply:
but why arent you passing the details of the TX when it is confirmed? the global tx queue frequently polls the status of all pending txs?

Reply:
For that part, we have to go to the state/transactions/updater

there the transactions in-store are polled, and there I have the details of the receipt.
or did you mean something else?

Reply:
so flow now is: SwapConfirmation > Buy > Onsent > Updater(shouldCheck/Updater) > Finalized > popup